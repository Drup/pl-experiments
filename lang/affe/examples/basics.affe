let id x = x
let app f x = f x
let ignore x = ()
let plus x y = x + y