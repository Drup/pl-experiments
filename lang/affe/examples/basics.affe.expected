id : ∀^k, 'a. ('a : ^k) => 'a -> 'a

app :
  ∀^k, ^l, ^m, 'a, 'b. ('b : ^m) & ('a : ^l) =>
  ('a -{^k}> 'b) -> 'a -{^k}> 'b

ignore : ∀^k, 'a. ('a : ^k) & (^k < affₙ) => 'a -> unit

plus : int -> int -> int

